src/main/java/com/deloitte/elrr/aggregator/utils/ExtensionsUtil.java

```diff
@@ -0,0 +1,134 @@
+package com.deloitte.elrr.aggregator.utils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.stereotype.Component;
+
+import com.deloitte.elrr.aggregator.rules.ExtensionsConstants;
+import com.deloitte.elrr.elrraggregator.exception.AggregatorException;
+import com.deloitte.elrr.entity.types.GoalType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yetanalytics.xapi.model.AbstractActor;
+import com.yetanalytics.xapi.model.Activity;
+import com.yetanalytics.xapi.model.Context;
+import com.yetanalytics.xapi.util.Mapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class ExtensionsUtil {
+
+    /**
+     * @param context
+     * @param returnObject
+     * @return extensionsMap or LocalDateTime
+     * @throws AggregatorException
+     */
+    public Object getExtensions(Context context, String returnObject)
+            throws AggregatorException {
+
+        Map<URI, Object> extensionsMap = new HashMap<>();
+
+        try {
+
+            // Map
+            if (context != null && context.getExtensions() != null
+                    && returnObject.equalsIgnoreCase("Map")) {
+
+                extensionsMap = context.getExtensions().getMap();
+                return extensionsMap;
+
+                // LocalDatewTime
+            } else if (context != null && context.getExtensions() != null
+                    && returnObject.equalsIgnoreCase("LocalDateTime")) {
+
+                String strLocalDateTime = (String) context.getExtensions().get(
+                        ExtensionsConstants.CONTEXT_EXTENSIONS_EXPIRES);
+
+                if (strLocalDateTime != null) {
+                    return LocalDateTime.parse(strLocalDateTime,
+                            DateTimeFormatter.ISO_DATE_TIME);
+                }
+
+                // Actor
+            } else if (context != null && context.getExtensions() != null
```

Comment:
This section is particularly problematic because you are assuming that just because I want ONE key to be an actor that the context is full of actors. You can see how the approach above would clarify all of this and reduce a lot of this code and ambiguity. And you had the key from the documentation, so not sure why we would convert all extensions like this.